http-netty/src/main/java/io/micronaut/http/netty/NettyMutableHttpResponse.java

```diff
@@ -165,63 +264,38 @@ public HttpStatus getStatus() {
     @SuppressWarnings("unchecked")
     @Override
     public <T> Optional<T> getBody(Argument<T> type) {
-        return convertedBodies.computeIfAbsent(type.getType(), aClass -> getBody().flatMap(b -> {
-            ArgumentConversionContext<T> context = ConversionContext.of(type);
-            if (b instanceof ByteBuffer) {
-                return conversionService.convert(((ByteBuffer) b).asNativeBuffer(), context);
-            }
-            return conversionService.convert(b, context);
-        }));
+        return bodyConvertor.convert(type, body);
     }
 
     @Override
     public MutableHttpResponse<B> status(HttpStatus status, CharSequence message) {
         message = message == null ? status.getReason() : message;
-        nettyResponse.setStatus(new HttpResponseStatus(status.getCode(), message.toString()));
+        httpResponseStatus = new HttpResponseStatus(status.getCode(), message.toString());
         return this;
     }
 
-    /**
-     * @return The Netty {@link FullHttpResponse}
-     */
-    public FullHttpResponse getNativeResponse() {
```

Comment:
@dstepanov @graemerocher Removing this public method breaks binary compatibility between patch versions. This makes it extremely difficult for external libraries to provide compatibility across at least different patch releases of the same minor version.

I know this is an internal class, but still, couldn't we leave it there marked as deprecated and returning `toFullHttpResponse()` instead? I believe the `replace(ByteBuf)` method may also suffer from the same issue.

Btw, the reason I needed to use this method was that this class does not override the `getReason()` method, thus I saw no other option to get any previously set custom reason phrase.

Reply:
> the reason I needed to use this method was that this class does not override the getReason() method

The thing to do would have been to file an issue to get your use case supported instead of relying on internal behavior. Internal classes are not part of the public API and treating them as such would defeat the purpose of having them in the first place

Reply:
@lgathy I understand your pain, binary compatible might have been worse solution here as semantics changed and you aren’t getting stored instance but a new one. The best solution is to create a public api interface with netty specific methods.

Reply:
@jameskleeh you are absolutely right about raising an issue, I will do that. I understand using internal API is dangerous, in this instance, I had no other choice as I needed to support older Micronaut versions too.

Reply:
Raised an issue with an example app to demonstrate the current behaviour: https://github.com/micronaut-projects/micronaut-core/issues/5544